### ISSUE TITLE
Make `Criterion` implementations accept logits as input

### ISSUE THREAD

**Issue by WenjieDu** [WenjieDu/PyPOTS#655]
### 1. Feature description

Currently the input of forward() is `prediction` and `target`, but `prediction` is quite fuzzy
https://github.com/WenjieDu/PyPOTS/blob/47080186912db2e58dee4663cb58ecd97144c90a/pypots/nn/modules/loss.py#L22-L27

### 2. Motivation

Now that `Criterion` is a subclass of torch.nn._Loss and torch.nn.Module, it and its subclasses are supposed to take outputs from torch.nn.Module. Use `logits` can unify their inputs.

### 3. Your contribution

Will make a PR
